tests/e2e/demo-modal-compare.spec.ts

```diff
@@ -0,0 +1,100 @@
+  if (!packageName) return;
+
+  const formattedPackageName = packageName
+    .split('-')
+    .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
+    .join(' ');
+
+  test.describe(`Demo ${formattedPackageName} Components`, () => {
+    test(`test demo ${formattedPackageName} component ${componentName}`, async ({ page }) => {
+      try {
+        const url = isTestingEnvironment()
+          ? SERVERS.TESTING[packageName]
+          : getDevelopmentEndpointUri(packageName, { isDocker: packageName !== 'web-twig' });
+
+        await page.goto(`${url}${componentsDir}/${componentName}/${packageName === 'web-twig' ? '?HIDE_TOOLBAR' : ''}`);
+        await waitForPageLoad(page);
+
+        await runModalTests(page, componentName);
+      } catch (error) {
+        console.error(`Test for demo ${formattedPackageName} component ${componentName} failed. ${error}`);
+        throw error;
+      }
+    });
+  });
+};
+
+const waitForPageLoad = async (page) => {
+  // Wait for fonts to load
+  await page.evaluate(() => document.fonts.ready);
+
+  // Wait for images to load
+  await page.waitForFunction(() => Array.from(document.querySelectorAll('img')).every((img) => img.complete));
+
+  // Wait for transitions to finish
+  await page.waitForLoadState();
+
+  // Disable animations to avoid flaky screenshots
+  await page.addStyleTag({ content: '*, *::before, *::after { animation-iteration-count: 1 !important }' });
+};
+
+const runModalTests = async (page, componentName) => {
+  // Helper function for taking and comparing screenshots
+  const takeScreenshot = async (suffix, fullPage = false) => {
+    await expect(page).toHaveScreenshot(`${componentName.toLowerCase()}-${suffix}.png`, {
+      animations: 'disabled',
+      fullPage,
+    });
+  };
+
+  // TEST: open basic modal, close with backdrop click
+  await page.click('[data-test-id="basic-modal"]');
+  await takeScreenshot('basic');
+  await page.locator('body').click({ position: { x: 0, y: 0 } });
+
+  // TEST: open modal and dropdown inside, close with close button
+  await page.click('[data-test-id="modal-with-dropdown"]');
+  await page.click('dialog[open] .Dropdown button');
+  await takeScreenshot('with-dropdown');
+  await page.click('dialog[open] .ModalHeader button');
+
+  // TEST: open stacking modals, close with escape key
+  await page.click('[data-test-id="modal-stacking"]');
+  await page.click('dialog[open] .ModalFooter button');
+  await takeScreenshot('stacking');
+  await page.keyboard.press('Escape');
+  await page.keyboard.press('Escape');
+
+  // TEST: open modal with hidden close button, close with action button
+  await page.click('button[data-test-id="hidden-close-button"]');
+  await takeScreenshot('hidden-close-button');
+  await page.click('dialog[open] .ModalFooter button');
+  await takeScreenshot('closed-modals', true);
+};
+
+const testConfigs = [
+  {
+    componentName: 'Modal',
```

Comment:
do we need this?

Reply:
I at least put it in a constant so the string doesn't repeat.